src/hooks/useWatchBundle.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  iterateEach,
+  makeCastingSpec,
+  makeFollower,
+  makeLeader,
+} from "@agoric/casting";
+import { toast } from "react-toastify";
+import { useNetwork } from "./useNetwork";
+import { BundleFollowerToastMessage } from "../components/BundleFollowerToastMessage";
+
+type IteratorEnvelope = {
+  value?: {
+    endoZipBase64Sha512: string;
+    installed: boolean;
+    error: unknown;
+  };
+  error?: { message: string; stack: string };
+};
+
+export const useWatchBundle = () => {
+  const { networkConfig } = useNetwork();
+  const leader = networkConfig ? makeLeader(networkConfig.rpc) : undefined;
```

Comment:
hm... I think I enhanced casting to accept an RPC client object, i.e. explicit access to the network, and not just a URL that gets turned into network access like casting an int to a pointer.

Or maybe I just figured out how to.

Reply:
[`makeHttpClient`](https://github.com/Agoric/agoric-sdk/commit/5bc513853d144f2967d9c2fc5b21fa2c4e284982) ?

It seems I need a follower to be able to query `:bundles`? I'm not sure how [this](https://github.com/Agoric/agoric-sdk/blob/af57c38c8d25498d8fa0261349b2c7647422d914/packages/casting/src/casting-spec.js#L66) maps wrt things like `:1317/agoric/(swingset|vstorage)/*` and `agd query (vstorage|swingset)... `.